src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef, useState } from "react";
+import ArrowDownIcon from "../icons/ArrowDownIcon";
+import styles from "./Dropdown.module.scss";
+
+export type Option = {
+  key: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  /** Массив возможных вариантов для выбора */
+  options: Option[];
+  /** Текущий индекс, может быть пустым */
+  keySelected: string;
+  /** Callback, вызываемый при выборе варианта */
+  onChange: (key: string) => void;
+};
+
+const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  keySelected,
+  onChange,
+}: DropdownProps) => {
+  let [isExpanded, Toggle] = useState(false);
```

Comment:
```suggestion
  let [isExpanded, setIsExpanded] = useState<boolean>(false);
```
так понятнее, что это функция, изменяющая state